### PR TITLE
[ADD] account_generator: generate dummy data

### PR DESCRIPTION
**Pull request by william-andre**
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mreficent** [comment]
Couldn't this be merged in `account_test`? 🤔

File: addons/account_generator/models/generator.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        line_ids, view = self._create_records('account.move', create_vals)
+        if self.post:
+            line_ids.post()
+            _logger.info('Journal Entries posted')
+        return view
+
+    def generate_partners(self):
+        create_vals = []
+        for i in range(self.number_to_generate):
+            create_vals.append({
```
Comment (gdgellatly) [inline addons/account_generator/models/generator.py]: would be nice to see all the val acquisitions split out to _prepare_* methods if this is for any kind of demo usefuleness

**Comment by pedrobaeza** [comment]
Coming to this through codetriage.com. Is this still relevant?

**Comment by william-andre** [comment]
It is put on the side ATM but not abandoned

File: addons/account_generator/models/generator.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        bank_statement_ids, view = self._create_records('account.bank.statement', create_vals)
+        return view
+
+    ###### Tools and Misc ######################################################
+
+    def _create_records(self, model_name, create_vals):
+        model = self.env[model_name]
+        _logger.info('Start Create')
+        model.check_access_rights('write')
+        self.env.su = True
```
Comment (oco-odoo) [inline addons/account_generator/models/generator.py]: Why not just doing a sudo() ?

File: addons/account_generator/models/generator.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    start_date = fields.Date(default=fields.Date.today() + datetime.timedelta(days=-365 * 3))
+    empty_field_probability = fields.Float(default=0.7)
+
+    company_id = fields.Many2one('res.company', default=lambda self: self.env.company)
+    partner_ids = fields.Many2many('res.partner', relation="generator_partner_pool")
+    account_ids = fields.Many2many('account.account', relation="generator_account_pool")
+    journal_ids = fields.Many2many('account.journal', relation="generator_journal_pool")
+    partner_excluded_ids = fields.Many2many('res.partner', relation="generator_excluded_partner")
+    account_excluded_ids = fields.Many2many('account.account', relation="generator_excluded_account")
+    journal_excluded_ids = fields.Many2many('account.journal', relation="generator_excluded_journal")
```
Comment (oco-odoo) [inline addons/account_generator/models/generator.py]: What's the point of these excluded journals and accounts? I don't see any real added value in using that.

File: addons/account_generator/models/generator.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                'date': self.end_date,
+            })
+        bank_statement_ids, view = self._create_records('account.bank.statement', create_vals)
+        return view
+
+    ###### Tools and Misc ######################################################
+
+    def _create_records(self, model_name, create_vals):
+        model = self.env[model_name]
+        _logger.info('Start Create')
```
Comment (oco-odoo) [inline addons/account_generator/models/generator.py]: Maybe also log the model name

File: addons/account_generator/models/generator.py (hunk 2 above)
Comment (oco-odoo) [inline addons/account_generator/models/generator.py]: sudo() ?

File: addons/account_generator/models/generator.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            'type': 'ir.actions.act_window',
+            'domain': [('id', 'in', record_ids.ids)],
+        }
+
+    def _random_or_empty(self, items):
+        if random.uniform(0, 1) > self.empty_field_probability or not len(items):
+            return False
+        return random.choice(items)
+
+    def _get_records(self):
```
Comment (oco-odoo) [inline addons/account_generator/models/generator.py]: We only call it once; so do we need this function ?

File: addons/account_generator/models/cleaner.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    _description = "Account Cleaner"
+
+    company_ids = fields.Many2many('res.company', default=lambda self: self.env.companies)
+
+    def clean_account_move(self):
+        company_ids = tuple(self.company_ids.ids)
+
+        self.env.cr.execute('delete from account_partial_reconcile where company_id in %s;', [company_ids])
+        self.env.cr.execute('delete from account_move_line where company_id in %s;', [company_ids])
+        self.env.cr.execute('delete from account_move where company_id in %s;', [company_ids])
```
Comment (oco-odoo) [inline addons/account_generator/models/cleaner.py]: Can't this cleaner stuff be a little too dangerous when ran on a production db? The simple fact people could install that themselves gets me worried... Or maybe with something prompting the user like "are you really really 100% super certain that you want to do that?". But even so, I find it dangerous.

**Review by oco-odoo — CHANGES_REQUESTED** [review]

**Comment by william-andre** [comment]
Not needed anymore since the populate mechanics

**Pull request closed without merging**